avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -675,6 +675,22 @@ private void processArray(JVar arraySchemaVar, final String name, final Schema a
     }
   }
 
+  /**
+   * Return a JExpression, which will read a string from decoder and construct a stringable object.
+   * @param stringbleClass
+   * @return
+   */
+  private JExpression readStringableExpression(JClass stringbleClass) {
+    JExpression stringableArgExpr;
+    if (Utils.isAvro14()) {
+      stringableArgExpr = JExpr.direct(DECODER + ".readString(null).toString()");
+    } else {
+      // More GC-efficient
+      stringableArgExpr = JExpr.direct(DECODER + ".readString()");
```

Comment:
can you explain here why avro 1.5+ can be treated differently?

Reply:
This function is not  available in avro-1.4 and available in 1.7/1.8 which are versions supported by fast-avro.
I could add some comment for this.